docs/book/adapter/dbtable/credential-treatment.md

```diff
@@ -3,8 +3,30 @@
 `Zend\Authentication\Adapter\DbTable\CredentialTreatmentAdapter` will execute a
 SQL query containing the provided identity and credentials, passing the
 credentials to a *credential treatment* function defined on the RDBMS server;
-if an identity is returned, authentication succeeds. Common credential
-treatments include `MD5()` and `PASSWORD()`.
+if an identity is returned, authentication succeeds. Credential
+treatments depends on your RDBMS, and while simple hashing function such as
+`md5` and `sha1` are generally available, it is recommended not to use them and
+rather use the RDBMS specific function such as
+[`PASSWORD(?)` for MySQL](http://dev.mysql.com/doc/refman/5.7/en/password-hashing.html) or
+[`crypt()` for PostgreSQL](https://www.postgresql.org/docs/11/pgcrypto.html#id-1.11.7.34.6).
+More details are available in the next section.
+
+## Security considerations
+
+Passing passwords to database in plaintext for insert or verification is
+generally not recommended.  
+Sql statements can and usually are logged by the database, passwords in them
+become visible to anyone with access to the logs or monitoring tools that
+consume those logs.
+
+Safer way is to hash passwords and verify them against stored hash on the
+application side. This way password never needs to leave application and only
+hashed value exchanged with database.
+
+As such, this adapter is not recommended for new applications and existing
+applications should consider migrating to using PHP provided password handling
+functions `password_hash()` and `password_verify()`. See
```

Comment:
Replace starting at `using` and ending at `functions` with the following: `using PHP-provided password handling functions such as`